### PR TITLE
Update EIP-7928: Clarify edge case aorund 7702 insufficient gas

### PR DESCRIPTION
**Pull request by nerolation**
Clarify an edge case around 7702.
Test case:
https://github.com/ethereum/execution-specs/pull/2824

### CODE REVIEW

**Review by fselmo — APPROVED** [review]
lgtm 👍🏼

For reference:

- recent `execution-specs` change [here](https://github.com/ethereum/execution-specs/pull/2473)
- test case expectation amended [here](https://github.com/ethereum/execution-specs/pull/2882)

File: EIPS/eip-7928.md [hunk 1]
```diff
@@ -164,7 +164,13 @@ Post-state costs (e.g., `GAS_NEW_ACCOUNT` for calls to empty accounts, `GAS_SELF
 
 #### EIP-7702 Delegation
 
-When a call target has an [EIP-7702](./eip-7702.md) delegation, the target is accessed to resolve the delegation. If a delegation exists, the delegated address requires its own `access_cost` check before being accessed. If this check fails, the delegated address MUST NOT appear in the BAL, though the original call target is included (having been accessed to resolve the delegation).
+When a call target has an [EIP-7702](./eip-7702.md) delegation, the target is accessed to resolve the delegation. The delegated address is loaded, and MUST appear in the BAL, only if **all** of the following hold:
+
+- Sufficient gas remains to cover the delegated address's own `access_cost`.
+- For value-transferring `CALL` or `CALLCODE`, the sender's balance covers the value transfer (`sender_balance >= value`).
+- The call stack depth limit is not violated.
```
Comment (jochem-brouwer) [inline EIPS/eip-7928.md:171]: So in case of a CALL/CALLCODE with value > balance, and there is sufficient gas to pay for rearching TARGET, this would then add TARGET to the BAL. And then the actual value check / gas check / depth check happens, if this pass we also add the DELEGATE to the BAL, because this is the "moment" we are actually sure we have to read it so it is necessary to add?
```
name | CONTRACT -----> TARGET ----------> DELEGATE
code | 0xc0dec0de ----> 0xef0100de... --> DELEGATE
```

I found this slightly confusing and counter-intuitive but the more I think about it, the more it makes sense and looks correct.

Comment (nerolation) [reply EIPS/eip-7928.md:171]: Yep, that's exactly it, TARGET gets added once we've paid to reach it, and DELEGATE only gets added once value/gas/depth all clear, since that's the first point we're committed to actually loading it
